doc/procedures/pre_commit.rst

```diff
@@ -31,11 +31,24 @@ These configuration files facilitate ``pre-commit`` to maintain ``black`` format
 If ``mypy`` is used, a configuration file can be generated for that as well.
 The configuration files are generated by using the ``generate_pre_commit_conf`` command.
 Since TSSW uses ``conda`` (or ``mamba``) for their software installations, a conda recipe was created for this project and the conda package was made available in the ``lsstts`` conda channel.
-To enable this:
 
-* Install pre-commit, for instance ``conda install -y pre-commit``.
-* Install ``ts_pre_commit_conf``, for instance ``conda install -y -c lsstts ts-pre-commit-config``.
-* Then execute the ``generate_pre_commit_conf`` command.
+To enable this in a new project or a project that isn't configured for pre-commit yet, follow the instructions for installing pre-commit in the previous section.
+In case of a project that already has a ``.pre-commit-config.yaml`` configuration file, execute these steps:
+
+* Inspect the ``.pre-commit-config.yaml`` file to see if the project uses mypy or not.
+* Remove the ``.pre-commit-config.yaml`` file with ``git rm --cached .pre-commit-config.yaml``.
+
+Then, in all cases, execute these steps:
+
+* Install ``ts_pre_commit_conf`` if not already done, for instance ``conda install -y -c lsstts ts-pre-commit-config``.
+* Then execute the ``generate_pre_commit_conf`` command with ``generate_pre_commit_conf --create`` or ``generate_pre_commit_conf --create --no-mypy`` if mypy isn't used.
+* Add the newly created ``.ts_pre_commit_config.yaml`` to git with ``git add .ts_pre_commit_config.yaml``.
+* Committing the changes at this stage may lead to errors from the pre-commit hooks, so first run pre-commit on the command line with ``pre-commit run --all-files``.
+  If this changes anything, then fix as needed.
+  Changes made by black can be accepted "as is".
+  Changes made by isort are usually innocuous, but it is configured to sort includes in ``__init__.py`` files, so there is potential for breakage.
+  If mypy complains, update your code.
+* Once this is all done, create a git commit to reflect the change with ``git commit -a -m "Use ts_pre_commit_conf"``.
```

Comment:
Is the info below (line 53 and on) is still needed? It surprised me to see that right after these instructions. Please consider moving up above these new instructions, or perhaps deleting it.

Reply:
Thanks for spotting that. I think the section still contains useful information so I moved it up above the new instructions.